### DISCUSSION TITLE
Qual é sua dica de produtividade ?

### DISCUSSION THREAD

**Discussion by romulosardinha** [frontendbr/forum#1369] [Discussão]
Cada vez mais me apaixono por produtividade e como posso melhorar sempre.
- Ouvir músicas calmas ao invés de agitadas
- Manter os pés altos para não causar incômodo na cadeira
- Manter o ambiente mais escuro que minha tela
- Altura da mesa e teclado 
- Usar Componentes + Snippets ( esta recentemente está salvando horas )

E com vocês? Quais detalhes fazem vocês trabalharem melhor, mais rápido ou mais felizes ?

**Comment by ghost** [comment]
Gostei das dicas, mas fiquei em dúvida em uma delas, manter o ambiente mais escuro que a tela, ou até mesmo ficar na frente do computador no escuro, isso não causa problemas de visão não? Não seria melhor ao contrário, deixar o ambiente mais claro e a tela com brilho menor?

**Comment by romulosardinha** [comment]
@marcosabb, pelo contrário, diminuo a iluminação do ambiente, cortinas, e luzes de led apenas daquelas escondidas sabe, abaixo da mesa, cantos do teto... e a tela fica com brilho bem baixo, já ajuda.
antes eu trabalhava de frente para uma janela, ai tinha a luz do dia + a luz da tela, ficava com dores nos olhos algumas vezes.

**Comment by ghost** [comment]
@rmlSardinha entendi, massa!

**Comment by eduardojmatos** [comment]
Eu gosto de pensar que cada um tem uma forma de trabalhar de forma mais produtiva. Eu realmente gostava mais de trabalhar em luz baixa, mas eu não era tão produtivo - pelo menos era o que eu sentia.

O lance é saber como medir isso, pra saber se não há falsos positivios de produtividade. Nem sempre a relação entre trabalhar num ambiente confortável está ligada diretamente ao aumento de produtividade. Felicidade talvez sim, mas de repente vale alguma maneira de medir isso - e sempre algo que você faça consigo mesmo, nada de processos de empresa, a não ser que seja algo benéfico e saudável (medir número de linhas de código feitas, por exemplo, não é nada saudável).

**Comment by luizbills** [comment]
Uma coisa que faz total diferença para mim é **saber o que devo fazer**. Eu costumava dar uma pensada bem rápida no projeto/tarefa e em seguida botar a mão na massa. Para projetos pequenos isso nunca deu problema. Mas quando era projetos maiores, muitas vezes eu me via mudando/refatorando coisas que já estavam prontas mas que precisavam melhorar.

Hoje em dia, eu gosto de gastar um bom tempo *planejando* o projeto. Quebro em várias tarefas e geralmente essas tarefas ainda são quebradas em micro tarefas. Planejo estudos que serão necessários, já testo pacotes que ajudaram no desenvolvimento, etc.

Descobri que uma chave para minha produtividade é a **organização**.

**Comment by erickmajor** [comment]
Aproveitando o quê o @luizbills já disse, eu ainda incremento que para ter mais produtividade, podemos utilizar ferramentas como o Kanban para termos uma dimensão do andamento do desenvolvimento. Uma ferramenta que eu utilizo bastante é o [Trello](https://trello.com/).

Uma ferramenta bem simples e que me ajuda a manter o foco é a Técnica Pomodoro. Com ela consigo manter meus momentos de foco com mais produtividade. Um app que me auxilia bastante é o [Brain Focus](https://play.google.com/store/apps/details?id=com.AT.PomodoroTimer).

Quanto ao ambiente, acredito que sendo ergonômico, é satisfatório para o desenvolvimento. Vai de cada um a personalização. =D

**Comment by romulosardinha** [comment]
@erickmajor eu já usei o trello, mas não me ajudou muito, na verdade o [habitica ](https://habitica.com) me ajudou mais a organizar todas as tarefas dos diferente clientes.

**Comment by erickmajor** [comment]
@rmlSardinha eu também uso o Habitica! No meu caso, eu utilizo para criar hábitos (p. ex., estudar a linguagem **Go** todos os dias às 20h).

A questão que expus é sobre a técnica do Kanban. Ela me ajuda muito a verificar o andamento dos meus projetos. Fique a vontade de escolher a ferramenta que lhe agrada mais. =D

Quando me referi ao Trello, para mim, é mais para se ter uma noção do andamento do projeto. Para ter uma dimensão do aproveitamento de tempo e onde eu devo me dedicar mais para atingir os prazos. A visão das piscinas e os cartões, que representam as micro tarefas, me ajuda a ter essa dimensão de forma visual, pq eu sou uma pessoa mais visual.

Existem várias técnicas de gerenciamento de tarefas disponíveis. Basta encontrar uma que lhe agrade. =D

**Comment by luizbills** [comment]
@erickmajor bem lembrado. Também uso o trello há anos, mas nem lembrei de citar ele kkk

Pomodoro também é muito util, ainda mais se você for freelancer. Com o tempo, pomodoro meio que vira uma unidade de tempo e te ajuda muito a saber o tempo necessário para uma tarefa

**Comment by romulosardinha** [comment]
Meditar pelo menos 5 min antes de iniciar a codar.

**Comment by romulosardinha** [comment]
@eduardojmatos , uso como medida pessoal a facilidade que consigo chegar aquele estado de flow.
como eu disse, músicas agitadas podem até ser boas no inicio, mas não me fazem mergulhar na concentração em longo do período. 

ter café e bastante água tbm ajuda muito nisso. utilizar o pomodoro está sendo um aprendizado ainda, não sei dizer ao certo se senti diferença com ele.

estou esperando um monitor ultra wide chegar para ver se tenho ganhos tbm com ele.

**Comment by didiraja** [comment]
Pra mim sempre será alguém falando no ouvido, seja podcast ou algum programa de Youtube

**Comment by felipeporto** [comment]
Legal essa proposta do Habitica, vou testar.

**Comment by felipefialho** [comment]
@anabastos vi você perguntando no twitter esses dias :)

**Comment by wmsbill** [comment]
Já que ninguém postou, aqui vai a minha contribuição: https://www.noisli.com/

**Comment by valterbarros** [comment]
@wmsbill caraca mano esse esse site é interessante.

**Comment by romulosardinha** [comment]
@wmsbill gostei muito desse ai em

**Comment by romulosardinha** [comment]
eu estava usando esses dias uma extensão do vscode que deixava marcações no código, que eu posso navegar de uma para outra de maneira rápida, apenas no comando, mas parece que buga um pouco, as vezes só por ficar editando o código as marcações pulam uma ou duas linhas... vou ver se tem algo na configuração. Mas se quiserem experimentar.

https://marketplace.visualstudio.com/items?itemName=alefragnani.Bookmarks

**Comment by wmsbill** [comment]
Sempre uso parra entrar na matrix. Mas não cometa o mesmo erro que eu. Se quiser comprar o app e tiver iphone X pra cima, ele não funfa..

**Comment by ninetails** [comment]
Que me ajudou por um tempo foi fazer sempre no final do dia (expediente ou antes de dormir) uma [matriz de Eisenhower](https://www.eisenhower.me/eisenhower-matrix/), atualizando e jogando fora tasks que não faziam mais sentido.

Pomodoro nunca deu certo pra mim.

De resto, diria escolher horários alternativos para trabalhar... À noite geralmente tem menos interrupções, barulho e trânsito (pra quem tem horário flexível mas não home office).

Já em programação, acho que automatizar e botar Git Hooks pra:
- não commitar código sem padrão (linter & prettier)
- não subir código quebrando teste

mesmo com um CI integrado com o git na hora do Pull Request, antes cada dev ter que corrigir seu código quando ainda a lógica está fresca na cabeça a que ter que corrigir depois do CI apitar e estar trabalhando em outra task ou outro ter que corrigir.

**Comment by joaoricardofl** [comment]
Acho que esse vídeo tem umas dicas interessantes para aumentar a produtividade: https://www.ted.com/talks/shawn_achor_the_happy_secret_to_better_work#t-662518
Ele fala principalmente de psicologia positiva fazendo ligação com a produtividade, e uma das recomendações que tem no vídeo e que faz uma boa diferença pra mim, é fazer exercícios regularmente, me deixa com mais disposição pra trabalhar.

**Comment by simaodeveloper** [comment]
- Fazer pausas de 20 segundos a cada 10 min para alongar o corpo;
- a cada 60 min uma pausa mais longa de 5 a 10 min, para caminhar, comer uma fruta, olhar pro nada rs

As vezes precisamos nos retirar de um contexto por algum momento para voltar com foco total.

**Comment by devGuerra** [comment]
Eu estou tentando usar uma:

- Fazer uma coisa por vez, ficar mudando de foco toda hora prejudica muito a produtividade

**Comment by caioincau** [comment]
Exato, multi-tasking é totalmente improdutivo, recomendo o livro "O Gestor Eficaz" lá tem várias dicas de produtividade, dicas reais com base em estudos, não achismos de coach.

**Comment by expalmer** [comment]
@ninetails Você usa algum programa para matriz de Eisenhower ? tem um exemplo ?

**Comment by ninetails** [comment]
@expalmer da primeira vez vi nesse [app pra android](https://play.google.com/store/apps/details?id=com.andtek.sevenhabits), mas depois comecei a fazer na mão mesmo, cada dia uma folha de caderno e eu ia atualizando e mudando as prioridades do que acabava sobrando... também deixava um "Parking Lot" com coisas não tão importantes pra não ficar "sujando" a matriz, também ia despriorizando e jogando lá o que via que não era mais tão importante ou apagava se era totalmente inútil.

**Comment by tarciozemel** [comment]
Dicas de Jules Payot, em "A Educação da Vontade":

> Para agir, é preciso realizar corajosamente cada uma das pequenas ações que concorrem para alcançar uma finalidade.

No que complementa com essa listinha de dicas práticas:

- Fixar, toda noite, as tarefas do dia seguinte
- Terminar todo o trabalho começado
- Fazer uma só coisa de cada vez
- Não desperdiçar nenhuma parcela do seu tempo

> A paciência dos esforços incessantemente renovados produz resultados prodigiosos: é, portanto, o hábito da atividade incessante que o estudante deve adquirir.

**Comment by degaki** [comment]
Eu escrevi recentemente algumas dicas que peguei de livros que li sobre organização e produtividade. Digo que GTD (Get Things Done) e o Bullet Journal são os que mais contribuíram para melhorar minha produtividade. Quem tiver interesse pode ver em: [https://www.quicktip.com.br/5-livros-para-ajudar-a-organizar-o-seu-dia/](url)

**Comment by leandrocrs** [comment]
### Som ambiente (sem voz preferencialmente)
Te ajuda a entrar em flow e inibe ruídos externos. Meus estilos preferidos: lo-fi, jazzhop, chillhop, down-tempo, chillstep (quando é sexta e o ritmo é mais animado).

### Fones anti-ruído
São recomendados e ajudam a preservar a audição;

### Usar um copo ao invés de garrafa
Te obriga a levantar a bunda da cadeira frequentemente para buscar água, ajudando na circulação sanguínea nas pernas;

### Realizar boas pausas
Cultive o hábito de realizar pausas mais longas; vá para uma área externa ou em uma janela e tente focar em algo distante. Previne a _cegueira focal_, causada por longas sessões olhando para uma tela, e também dá tempo para o _modo difuso_ do cérebro entrar em ação. Já aconteceu de você encontrar a solução pra um problema enquanto estava tomando banho ou no banheiro? Pois é, não foi a cagada ou a água do chuveiro, e sim seu cérebro resolvendo o problema em segundo plano (modo difuso).

### Reduza a cafeína depois das 16hs
A cafeína leva em média 10hs para sair do corpo. Se você beber café durante a tarde, provavelmente estará sob o efeito dela durante a noite, o que poderá reduzir a duração e/ou qualidade do sono. Isso vai te prejudicar muito no dia seguinte.

### Não exagerar no almoço
Com o estômago pesado, o corpo foca em digerir tudo o que você comeu, te deixando sonolento. Coma o suficiente e não mais que isso, seu corpo vai criar reserva em gordura ou até mesmo descartar o excedente, então é perda de tempo e recursos comer mais do que deveria.

### Cada dia é um dia
Nem sempre você será produtivo. Tem dias que não dá, e tudo bem. Foque em algo que consiga acabar com certa facilidade para te dar uma animada. Se nem isso bastar, tudo bem. Faça o que pode, e nos dias que tiver se sentindo muito produtivo, mande ver, não se contenha.

### Mantenha o equilíbrio
Tente se limitar ao seu período de trabalho. Não faça mais do que 2 horas extras. Código é só o resultado final de todo o trabalho que acontece dentro da sua cabeça, então cuide muito bem dela. E evite na medida do possível de ficar mexendo com código o dia todo. É sério, mesmo que seja pra estudar algo ou praticar alguma coisa, tente não ficar o dia todo nisso. Chega uma hora que dá nojo de ver código.
> "O segredo do sucesso é a moderação" - Clube do Canalhas, Matanza

### Exercícios regulares
Faça exercícios moderados com frequência (estou tentando o 7 Minutes no android e estou gostando muito). Melhora o transporte de oxigênio pelo corpo, e te deixa cada vez mais disposto(a).

### Oxigene o cérebro
Leia, sério. Acho que das formas de consumo de mídia, é a que mais exige capacidade de abstração e imaginação.

### Não faça "nada"
Melhor coisa é descontrair, seja com o que for. E faça questão de ter espaço na agenda para fazer coisas que te deixam feliz. Seu cérebro precisa ser lembrado do porque você o submete todo dia a trabalhar com abstrações por várias horas. Dê um agrado à ele, seja o que for.

**Comment by romulosardinha** [comment]
@leandrocrs ótimo seu autoconhecimento cara. Descobrir como você funciona e qual objetivo quer alcançar com isso é tão valioso. E vejo Tão poucos fazendo. Quando comecei senti um retorno tão grande que não parei mais...

Vou fazer um adendo ao item do almoço: Não comam muito de uma vez, coma mais vezes durante o dia e coisas mais saudáveis. Isso desincha o abdômen. Nós somos o que comemos kk.

Uma que posso acrescentar recentemente. Tenha uma rotina antes do trabalho. Defina coisas que deseje fazer sempre antes do trabalho. Isso se torna um hábito e ajuda sua mente a se preparar a entrar em flow. 

E uma que sempre bato na tecla com todos que trabalham comigo. COMPREM BOAS CADEIRAS E MESAS!!! Vejo pessoas gastando mais de 60 mil no carro que não passam nem 1h nele por dia. Mas a cadeira e mesa que fica lá 10h ou mais, ta lá, todo torto. A postura da sua coluna,  pescoço curvado,  altura dos pés, altura da mesa para os braços não ficarem nem tão altos, nem tão baixos. Isso faz diferença tanto ao longo tempo como no curto tempo. Ao trabalhar em uma estrutura confortável é mais fácil focar e não ficar com aquela sensação de cansaço em 3h de código..

**Comment by eliseumds** [comment]
Puts, o @leandrocrs matou a pau. Vai experimentando coisas diferentes, como quantidade de luz, barulho, sono, almoço, fibras, café e etc e vai marcando o que te deixa melhor.
